src/hotspot/cpu/aarch64/aarch64_vector.ad

```diff
@@ -2854,11 +2854,8 @@ instruct reduce_addL_sve(iRegLNoSp dst, iRegL isrc, vReg vsrc, vRegD tmp) %{
 %}
 
 // reduction addF
-// Floating-point addition is not associative, so the rules for AddReductionVF
-// on NEON can't be used to auto-vectorize floating-point reduce-add.
-// Currently, on NEON, AddReductionVF is only generated by Vector API.
 instruct reduce_add2F_neon(vRegF dst, vRegF fsrc, vReg vsrc) %{
-  predicate(UseSVE == 0 && Matcher::vector_length(n->in(2)) == 2);
+  predicate(Matcher::vector_length(n->in(2)) == 2 && n->as_Reduction()->is_associative());
```

Comment:
This `vector_length(n->in(2)) == 2` is very obscure. I suspect that anyone coming across this code would not understand it.

What exactly is the reason that this pattern is only applied for the 16b case? You need to give a justification in a comment right here.

Reply:
This is for vector length of 8B (64 bits). It adds two floats. I have added my comments in the new PS.